libs/aws/langchain_aws/runnables/q_business.py

```diff
@@ -0,0 +1,166 @@
+
+    parent_message_id: Optional[str] = None
+
+    conversation_id: Optional[str] = None
+
+    chat_mode: str = "RETRIEVAL_MODE"
+
+    model_config = ConfigDict(
+        extra="forbid",
+    )
+
+    def __init__(
+        self,
+        region_name: Optional[str] = None,
+        credentials: Optional[Any] = None,
+        client: Optional[Any] = None,
+        application_id: str = None,
+        parent_message_id: Optional[str] = None,
+        conversation_id: Optional[str] = None,
+        chat_mode: str = "RETRIEVAL_MODE",
+    ):
+        self.region_name = region_name
+        self.credentials = credentials
+        self.client = client
+        self.validate_environment()
+        self.application_id = application_id
+        self.parent_message_id = parent_message_id
+        self.conversation_id = conversation_id
+        self.chat_mode = chat_mode
+
+    def invoke(
+        self,
+        input: str,
+    ) -> str:
+        """Call out to Amazon Q service.
+
+        Args:
+            input: The prompt to pass into the model.
+
+        Returns:
+            The string generated by the model.
+
+        Example:
+            .. code-block:: python
+
+            model = AmazonQ(
+                credentials=your_credentials,
+                application_id=your_app_id
+            )
+            response = model.invoke("Tell me a joke")
+        """
+        try:
+            # Prepare the request
+            request = {
+                'applicationId': self.application_id,
+                'userMessage': input,
+                'chatMode': self.chat_mode,
+            }
+            if self.conversation_id:
+                request.update({
+                    'conversationId': self.conversation_id,
+                    'parentMessageId': self.parent_message_id,
+                })
+
+            # Call Amazon Q
+            response = self.client.chat_sync(**request)
+            self._last_response = response
+
+            # Extract the response text
+            if 'systemMessage' in response:
+                return response["systemMessage"]
+            else:
+                raise ValueError("Unexpected response format from Amazon Q")
+
+        except Exception as e:
+            if "Prompt Length" in str(e):
+                logging.info(f"Prompt Length: {len(input)}")
+                print(f"""Prompt:
+                {input}""")
+            raise ValueError(f"Error raised by Amazon Q service: {e}")
+
+    def get_last_response(self) -> Dict:
+        """Method to access the full response from the last call"""
+        return self._last_response
+
+    async def ainvoke(
+        self,
+        input: str,
+    ) -> str:
+        """Async call to Amazon Q service."""
+
+        def _execute_call():
+            return self.invoke(input)
+
+        # Run the synchronous call in a thread pool
+        return await asyncio.get_running_loop().run_in_executor(
+            None, _execute_call
+        )
```

Comment:
I believe Runnable defaults to this behavior, so this might be redundant. Can we test after removing this?